isis/scripts/isis3VarInit.py

```diff
@@ -5,31 +5,68 @@
-#   Author:  Tyler Wilson
-#   Date  :  2018-10-05
-#   Description:  This script sets ISISROOT/ISIS3DATA/ISIS3TESTDATA for the user and is executed
-#   within the conda environment created for the ISIS3 installation.
-#   The data directory and test directory are optional command line arguments.  If the user chooses
-#   not to set them, they will both be placed created on the same level as the $ISISROOT directory
-#   within the conda environment.
+#
+#  This work is free of known copyright restrictions.  USGS-authored or produced data, 
+#  information, and software are in the public domain.
+#
+#
+#   Description:  This program builds the shell scripts that define the 
+#       ISISROOT/ISIS3DATA/ISIS3TESTDATA environment variables for the user 
+#       when the ISIS3 conda environment is activated, and clean up when it is
+#       deactivated.
+#
+#       The data directory and test directory are optional command line arguments.  
+#       If the user chooses not to set them, they will both be created in the
+#       $ISISROOT directory.
+#
 #   History:
-#       Author:  Tyler Wilson
+#       Author:  Tyler Wilson, USGS
+#       Date:    2018-10-05
+#       Description: Initial commit.
+#
+#       Author:  Tyler Wilson, USGS
 #       Date:    2018-11-01
 #       Description:  Removed a pair of lines which were causing output errors on Mac OS X and were not
 #                     required anyway.
 #
+#       Author:  Ross Beyer
+#       Date:    2018-11-
+#       Description: Streamlined the program, improved documentation, and made the directory and
+#                    file creation more `pythonic' rather than using system calls.
+#       To the extent possible under law, Ross Beyer has waived all copyright and related or 
+#       neighboring rights to his contribution to this file.  This work is published from
+#       the United States.
+#   
+#
 ##########################################################################################################
 
-parser = argparse.ArgumentParser(description='Usage:  ./isis3VarInit --data_dir <data dir path> --test_dir <test dir path')
-   
-isisroot = '$CONDA_PREFIX'
-data_dir='$CONDA_PREFIX/data'
-testdata_dir='$CONDA_PREFIX/testData'
+# There are still a lot of Python 2 installations out there, and if people don't have
+# their conda environment set up properly, the error message they'll get will be hard
+# to decipher.  This might help:
+assert( sys.version_info >= (3,2) ) # Must be using Python 3.2 or later, is conda set up?
+
 
-parser.add_argument("--data-dir",default= data_dir,help="ISIS3 Mission Data Directory")
-parser.add_argument("--test-dir",default=testdata_dir,help="ISIS3 Mission Test Data Directory")
+# This just wraps and reports on the directory creation:
+def mkdir( p ):
+    if os.path.exists( p ): print( 'Tried to create '+p+', but it already exists.' )
+    else:
+        os.makedirs( p )
+        print( 'Created '+p )
+    return
```

Comment:
I am confused about why this is needed. You can just use `exist_ok=true` on the [makedirs](https://docs.python.org/3/library/os.html?highlight=makedirs#os.makedirs) calls. Logging if the file exists already or not seems kind of pointless.

Reply:
You absolutely can, and that's the way I first wrote it (and why the assert  at the top is for 3.2, where exist_ok first shows up for makedirs).  However, after I did that, I realized that the program wasn't providing any feedback to the user about what the heck it was doing.  You just ran the program, and then it gave you a prompt, zero information for the user.  Did it work, did it quietly fail somehow?  Who knows?  So when I decided I wanted to provide some reporting about what it was doing, and I wanted to provide the user information about what already existed or not, that's when I made this little wrapper.  If you prefer the silent treatment, I can take it out.

Reply:
I'm fine with keeping this wrapper then